src/bin/pg_autoctl/pidfile.c

```diff
@@ -332,11 +327,14 @@ read_service_pidfile_version_strings(const char *pidfile,
 		/* extension version string, comes later in the file */
 		if (pidLine == PIDFILE_LINE_EXTENSION_VERSION)
 		{
+			free(fileContents);
```

Comment:
This `free` should happen after the `strlcpy` in the line below.

Reply:
Good point, done too!